deepseg.cc

```diff
@@ -213,6 +240,15 @@ int main(int argc, char* argv[]) {
 			} else {
 				showUsage = true;
 			}
+		} else if (strncmp(argv[arg], "-f", 2)==0) {
+			if (hasArgument) {
+				fourcc = fourCcFromString(argv[++arg]);
+				if (!fourcc) {
```

Comment:
Ambiguous regarding `0` vs. invalid. Use `std::optional` to disambiguate.

Reply:
I prefer optional myself, however:

* Are we on C++17? I don't see `-std=c++17` set in the Makefile and I'm not sure if Ubuntu 18.04 supports that.
* Zero is really invalid here. A fourcc of 0 doesn't make sense.

Reply:
We could use `absl::optional` even with C++11. It is used within tensorflow. That would, however, also require adapting the includes with the `Makefile` based build.

Reply:
Good point regarding the language standard (I'm used to C++17 and newer for my own projects) …

Iff 0 is really undefined/invalid this might be used for now, until the language standard issue is decided. Usually magic numbers should be avoided.

Reply:
> Usually magic numbers should be avoided.

I fully agree :)